parser.go

```diff
@@ -42,10 +42,13 @@ func (p *Parser) Parse(tokenString string, keyFunc Keyfunc) (*Token, error) {
 	return p.ParseWithClaims(tokenString, MapClaims{}, keyFunc)
 }
 
-// ParseWithClaims parses, validates, and verifies like Parse, but supplies a default
-// object implementing the Claims interface. This provides default values which
-// can be overridden and allows a caller to use their own type, rather than
-// the default MapClaims implementation of Claims.
+// ParseWithClaims parses, validates, and verifies like Parse, but supplies a default object implementing the Claims
+// interface. This provides default values which can be overridden and allows a caller to use their own type, rather
+// than the default MapClaims implementation of Claims.
+//
+// Note: If you provide a custom claim implementation that embeds one of the standard claims (such as RegisteredClaims),
+// make sure that a) you either embed a non-pointer version of the claims or b) if you are using a pointer, allocate the
+// proper memory for it before passing in the overall claims, otherwise you might run into a panic.
```

Comment:
This seems like a misuse of the language, no? I.e., the same could be said about adding to an uninitialized map in Go, that's how the language behaves and every package out there doesn't need to call it out.

Reply:
Yes it is, but since people seem to be running into this issue and this is a security-related package, it makes sense to point this out explicitly.